src/core/layertree/qgslayertreelayer.h

```diff
@@ -167,6 +168,13 @@ class CORE_EXPORT QgsLayerTreeLayer : public QgsLayerTreeNode
      */
     void setPatchShape( const QgsLegendPatchShape &shape );
 
+    /**
+     * Returns an expression representing the symbol associated with the \a ruleKey
+     *
+     * \since QGIS 3.14
+     */
+    QString symbolExpression( const QString &ruleKey );
```

Comment:
I think this particular function fits better in QgsLayerTreeUtils. Eg:


    /**
     * Returns an expression representing a legend key for a given layer tree \a node.
     * 
     * This method only works with nodes representing vector layers. For all nodes of all other layer types
     * an empty string will be returned.
     *
     * \since QGIS 3.26
     */
    static QString expressionForLegendKey( QgsLayerTreeNode* node, const QString& legendKey );